src/mesh/parallel/shiftedmetricinterp.cxx

```diff
@@ -124,7 +124,7 @@ ShiftedMetricInterp::ShiftedMetricInterp(Mesh& mesh, CELL_LOC location_in,
           it.ind, mesh.yend, z,
           mesh.GlobalX(it.ind),                           // x
           2. * PI * mesh.GlobalY(mesh.yend + 0.5),        // y
```

Comment:
Don't know if it's important, because I don't remember what `forward_boundary_xin`, etc., are for, but should the y-coordinate have a length that's not `2*pi` as well? `hypnotoad` now sets `dy` for a y-coordinate that goes from 0->2pi around the core so the total, global length of y is greater than 2pi if there are divertor legs: https://github.com/boutproject/hypnotoad/blob/2ea817c0094629c74c30101e495ff842280887de/hypnotoad/core/mesh.py#L3392-L3398

Reply:
I'm not convinced about these calculations of boundary point location. I suspect there is no "correct" way to do it at the moment, and an overhaul of the boundary system is needed.